src/task_userdefined.md

```diff
@@ -0,0 +1,148 @@
+
+example
+```yml
+rmf_fleet:
+  name: "ecobot40"
+  limits:
+    linear: [1.2, 1.5] # velocity, acceleration
+    angular: [0.6, 0.6]
+  profile: # Robot profile is modelled as a circle
+    footprint: 0.5
+    vicinity: 0.6
+  reversible: False
+  battery_system:
+    voltage: 24.0
+    capacity: 40.0
+    charging_current: 26.4
+  mechanical_system:
+    mass: 80.0
+    moment_of_inertia: 20.0
+    friction_coefficient: 0.20
+  ambient_system:
+    power: 20.0
+  cleaning_system:
+    power: 760.0
+  recharge_threshold: 0.05
+  recharge_soc: 1.0
+  publish_fleet_state: True
+  account_for_battery_drain: True
+  task_capabilities: # Specify the types of RMF Tasks that robots in this fleet are capable of performing
+    loop: True
+    delivery: False
+    clean: True
+    finishing_request: "park"
+    action_categories: ["clean", "manual_control"]
+
+```
+
+```python
+def _consider(description: dict):
+        confirm = adpt.fleet_update_handle.Confirmation()
+
+        # Currently there's no way for user to submit a robot_task_request
+        # .json file via the rmf-web dashboard. Thus the short term solution
+        # is to add the fleet_name info into action description. NOTE
+        # only matching fleet_name action will get accepted
+        if (description["category"] == "manual_control" and
+            description["description"]["fleet_name"] != fleet_name):
+                return confirm
+
+        node.get_logger().warn(
+            f"Accepting action: {description} ")
+        confirm.accept()
+        return confirm
+fleet_config = config_yaml['rmf_fleet']
+task_capabilities_config = fleet_config['task_capabilities']
+if 'action_categories' in task_capabilities_config:
+
+	for cat in task_capabilities_config['action_categories']:
+	
+		node.get_logger().info(
+		
+			f"Fleet [{fleet_name}] is configured"
+			
+			f" to perform action of category [{cat}]")
+	
+		fleet_handle.add_performable_action(cat, _consider)
+```
+2. [RobotUpdateHandle::set_action_executor](https://github.com/open-rmf/rmf_ros2/blob/8440488d5583edc5a5b7226326aa2a8d41dad975/rmf_fleet_adapter/include/rmf_fleet_adapter/agv/RobotUpdateHandle.hpp#L178-L179) This is where you tell the fleet adapter how to instruct your robot to begin performing an action. This callback to this function consists of 
+- `category(string)`  type of action
+- `description(JSON)`  message which contains details about how the action should be performed
+- `execution(object)` object which the platform-specific side of the fleet adapter must hold onto while the action is being performed, ideally giving periodic updates for remaining time estimates.
+
+The robot will not participate in the traffic negotiations while using a custom task in perform_action. That means that the robot will be allowed to report its trajectory to the traffic schedule thus making it possible for other robots to avoid the bot however the bot would not be able to accommodate other bots until the task is complete.
```

Comment:
```suggestion
The robot will not participate in the traffic negotiations while using a custom task in `perform_action`. That means that it will be allowed to report its trajectory to the traffic schedule, thus making it possible for other robots to avoid it. However, the robot would not be able to accommodate other robots until the task is complete.
```

Reply:
Let's not call it `bot`, I believe that term is mostly used for software programs.